include/envoy/http/filter.h

```diff
@@ -198,6 +198,16 @@ class StreamDecoderFilterCallbacks : public virtual StreamFilterCallbacks {
    * @param trailers supplies the trailers to encode.
    */
   virtual void encodeTrailers(HeaderMapPtr&& trailers) PURE;
+
+  /**
+   * Called when a decoder filter goes over its high watermark.
```

Comment:
Filter, codec, and connection for the router filter.  

Updating to (not upstream yet, still doing h2 config changes)  
   * Called when the buffer for a decoder filter or any buffers the filter sends data to go over
   * their high watermark.
   * In the case of a filter such as the router filter, which spills into multiple buffers (codec,
   * connection etc.) this may be called multiple times.  Any such filter is responsible for calling
   * the Low Watermark callbacks an equal number of times as the respective buffers are drained.